mypy/semanal.py

```diff
@@ -902,7 +902,9 @@ def analyze_base_classes(self, defn: ClassDef) -> None:
 
         base_types = []  # type: List[Instance]
         info = defn.info
-        for base_expr in defn.base_type_exprs:
+
+        base_type_exprs = self.check_with_metaclass(defn)
```

Comment:
I think this is not a right place for this check. It should be rather processed before (or in) ``clean_up_bases_and_infer_type_variables``, this is why you see problems with ``Generic`` in #3366.